### PR TITLE
Payment API documentation

### PR DESCRIPTION
**Pull request by vruizext**
Fixes https://github.com/socialpaymentsbv/billing-engine/issues/8114

### CODE REVIEW

**Comment by vruizext** [comment]
To see the file in a bit nicer format
https://github.com/socialpaymentsbv/docs.clubcollect.com/blob/b5df65b0ab8ea8e66196da8965cf8760f290942a/api-documentation/payments.md

**Review by cristianbica — APPROVED** [review]

**Review by pootsbook — APPROVED** [review]
LGTM :+1:

There are a number of English style improvements to make (e.g. “the request is not tampered” vs. “the request has not been tampered with”) but these can be made after the fact.

**Comment by vruizext** [comment]
> There are a number of English style improvements to make (e.g. “the request is not tampered” vs. “the request has not been tampered with”) but these can be made after the fact.

Fun fact, I copied this from Adyen's support page 😄

File: api-documentation/payments.md [hunk 1]
```diff
@@ -1,102 +1,445 @@
...
+`external_invoice_number` | `123456`
+`first_name` | `John`
+`last_name` | `Doe`
+`payment_reference` | `Club membership 2019/2`
+`redirect_url` | `http://partner-test.nl`
+
+3. Concatenate every key-value pair on the list obtained in the previous step to get the signing string
+
+```
+signing_string = params.keys.sort.flat_map { |k| [k, params[k]] }.join
```
Comment (gbrylinski) [inline api-documentation/payments.md:76]: or in a bit shorter way:

```ruby
params.sort.flatten.join
```

File: api-documentation/payments.md [hunk 2]
```diff
@@ -1,102 +1,445 @@
...
+
+##Payment response##
+
+Once the users finish the payment process, they are redirected to a result page of your choice, included in the payment request as `redirect_url`.
+
+ClubCollect will append parameters to this URL. `payment_result` will inform the partner about the payment status. If the status is already determined (either `authorised` or `refused` or `cancelled`), this information can be used to display a payment successful or payment failed or payment cancelled page.
+
+In a case when the current status is `pending`, the outcome of the payment will be communicated with payment notifications. Payments with status `cancelled` may be authorized later by the payment provider.
+
+The partner is encouraged to store the `invoice_id` and `payment_id` returned in ClubCollect responses and payment notifications, they will be needed to track the status of the payments. As an alternative, the partner can provide a unique `external_invoice_number` parameter in the payment request and ClubCollect will include it on the response and notifications.
```
Comment (gbrylinski) [inline api-documentation/payments.md:120]: it's probably to @pootsbook, I think it should be `include it in the response and notifications`.

**Review by gbrylinski — COMMENTED** [review]
👍

File: api-documentation/payments.md [hunk 3]
```diff
@@ -1,102 +1,445 @@
...
+After the signature is generated, it has to be appended to the url query string as
+`&signature=754966cc8946c8125b365fcb5cf0e27edd98fe7516de7ea17f1b5254bcf7a00e`
+
+##Payment response##
+
+Once the users finish the payment process, they are redirected to a result page of your choice, included in the payment request as `redirect_url`.
+
+ClubCollect will append parameters to this URL. `payment_result` will inform the partner about the payment status. If the status is already determined (either `authorised` or `refused` or `cancelled`), this information can be used to display a payment successful or payment failed or payment cancelled page.
+
+In a case when the current status is `pending`, the outcome of the payment will be communicated with payment notifications. Payments with status `cancelled` may be authorized later by the payment provider.
```
Comment (dcarneiro) [inline api-documentation/payments.md:118]: Here I would leave a reference saying that the cancelled payments might be authorised later via the payments api webhook.

**Review by dcarneiro — COMMENTED** [review]
I left one comment but this documentation seems pretty solid

**Pull request merged**